src/internal/symbol/iterator.ts

```diff
@@ -1,14 +1,19 @@
+import { root } from '../util/root';
 
-/* NOTE: Warning users that they don't have a Symbol.iterator
-   polyfill. We don't want to throw on this, because it's not required
-   by the library. However it will provide clues to users on older
-   browsers why things like `from(iterable)` doesn't work. */
-if (!Symbol || !Symbol.iterator) {
-  console.warn('RxJS: Symbol.observable does not exist');
+export function getSymbolIterator(root: { Symbol?: SymbolConstructor }): symbol {
+  /* NOTE: Warning users that they don't have a Symbol.iterator
+     polyfill. We don't want to throw on this, because it's not required
+     by the library. However it will provide clues to users on older
+     browsers why things like `from(iterable)` doesn't work. */
+  if (typeof root.Symbol !== 'function' || !root.Symbol.iterator) {
+    console.warn('RxJS: Symbol.iterator does not exist, so things like from(iterable) won\'t work');
+    return '@@iterator' as any;
+  }
+
+  return root.Symbol.iterator;
 }
 
-/** The native Symbol.iterator instance or a string */
-export const iterator = Symbol && Symbol.iterator || '@@iterator';
+export const iterator = getSymbolIterator(root);
```

Comment:
We actually want to get rid of `root` at some point. It's caused more issues than it's fixed.

Reply:
do you have alternatives for testing?